lang/py/test/test_schema.py

```diff
@@ -217,23 +218,72 @@ class InvalidTestSchema(ValidTestSchema):
 ]
 
 IGNORED_LOGICAL_TYPE = [
-  ValidTestSchema({"type": "string", "logicalType": "uuid"}),
-  ValidTestSchema({"type": "string", "logicalType": "unknown-logical-type"}),
-  ValidTestSchema({"type": "bytes", "logicalType": "decimal", "precision": 2, "scale": -2}),
-  ValidTestSchema({"type": "bytes", "logicalType": "decimal", "precision": -2, "scale": 2}),
-  ValidTestSchema({"type": "bytes", "logicalType": "decimal", "precision": 2, "scale": 3}),
-  ValidTestSchema({"type": "fixed", "logicalType": "decimal", "name": "TestIgnored", "precision": -10, "scale": 2, "size": 5}),
-  ValidTestSchema({"type": "fixed", "logicalType": "decimal", "name": "TestIgnored2", "precision": 2, "scale": 3, "size": 2}),
-  ValidTestSchema({"type": "int", "logicalType": "date1"}),
-  ValidTestSchema({"type": "long", "logicalType": "date"}),
-  ValidTestSchema({"type": "int", "logicalType": "time-milis"}),
-  ValidTestSchema({"type": "long", "logicalType": "time-millis"}),
-  ValidTestSchema({"type": "long", "logicalType": "time-micro"}),
-  ValidTestSchema({"type": "int", "logicalType": "time-micros"}),
-  ValidTestSchema({"type": "long", "logicalType": "timestamp-milis"}),
-  ValidTestSchema({"type": "int", "logicalType": "timestamp-millis"}),
-  ValidTestSchema({"type": "long", "logicalType": "timestamp-micro"}),
-  ValidTestSchema({"type": "int", "logicalType": "timestamp-micros"})
+  ValidTestSchema(
+    {"type": "string", "logicalType": "uuid"},
+    warnings=[schema.IgnoredLogicalType('Unknown uuid, using string.')]),
+  ValidTestSchema(
+    {"type": "string", "logicalType": "unknown-logical-type"},
+    warnings=[schema.IgnoredLogicalType('Unknown unknown-logical-type, using string.')]),
+  ValidTestSchema(
+    {"type": "bytes", "logicalType": "decimal", "scale": 0},
+    warnings=[schema.IgnoredLogicalType('Invalid decimal precision None. Must be a positive integer.')]),
+  ValidTestSchema(
+    {"type": "bytes", "logicalType": "decimal", "precision": 2.0, "scale": 0},
+    warnings=[schema.IgnoredLogicalType('Invalid decimal precision 2.0. Must be a positive integer.')]),
```

Comment:
Hmmmm.... I added this test, but I'm dubious. When precision is a fractionless floating point number, is the logical type valid or invalid? It currently is invalid and the logical type is ignored.

Reply:
Seems like the spirit of this part of the spec is postels law and we should make the best sense of inputs we can.

Reply:
I'll report back on what the Java implementation does...  (But it seems to me that they both should accept 2.0 as a precision; it's not like JSON has a float vs. int type!)

Reply:
Quick check with Java -- a floating point number (even without a fractional part) makes the decimal logical type invalid.  This was also the behaviour before this PR.

I propose leaving this behaviour ... if someone complains or feels strongly about it, we can revisit it at that point!

Reply:
Hmm, I suggest that we leave the behavior, but perhaps the test should be more to the point, and use something like 2.4?

Reply:
OK, done.